app/models/contact.rb

```diff
@@ -0,0 +1,23 @@
+class Contact < ApplicationRecord
+  validates :phone, uniqueness: true
+  validates :email, uniqueness: true
+
+  belongs_to :user
+
+  def display_name
+    name = ''
+    unless firstname.nil?
+      name << firstname
+    end
+    unless lastname.nil?
+      unless name == ''
+        name << ' '
+      end
+      name << lastname
+    end
+    if name == ''
+      name = '-'
+    end
+    name
+  end
```

Comment:
Nice to know: Ruby can interpolate strings:
```suggestion
  def display_name
    name = "#{firstname} #{lastname}".strip
    name.empty? ? '-' : name
  end
```

`strip()` removes leading and trailing spaces from strings...